cli/flox/src/commands/environment.rs

```diff
@@ -563,7 +510,7 @@ impl Activate {
             (flox_env_dirs, flox_env_lib_dirs)
         };
 
-        let shell = ShellType::detect()?;
+        let shell = Shell::detect()?;
```

Comment:
And I think `activate_in_place` should probably respect currently running shell over `FLOX_SHELL`. Less confident on that one 🤔

Reply:
> `activate_interactive` should respect SHELL over the currently running shell

Why should that respect SHELL over the actual shell? Wasn't the point of this to _not_ use SHELL if we know better?


> And I think activate_in_place should probably respect currently running shell over FLOX_SHELL.

This would remove the option for an explicit override though and in either case would be inconsistent with the other activations both current or modified as above.

Reply:
I view both of these variables more as something to put in an RC file (and honestly maybe we should move FLOX_SHELL to config), rather than something you set for a single invocation. I think you might be viewing it as more of a single command override?

> Why should that respect SHELL over the actual shell? Wasn't the point of this to _not_ use SHELL if we know better?

As I understand it, the point of this was fixing a bug in `activate_in_place`. There's no bug in `activate_interactive`. `SHELL` means that's the shell the user prefers. If I prefer zsh, happen to be in bash, and then run `flox activate`, there's no reason we shouldn't give the shell the user prefers.

> This would remove the option for an explicit override though and in either case would be inconsistent with the other activations both current or modified as above.

`FLOX_SHELL` means my favorite shell is `SHELL`, but flox doesn't support it, so when using `activate_interactive`, I prefer `FLOX_SHELL`. However, when I'm running a shell other than `FLOX_SHELL` and run `activate_in_place`, that doesn't mean I want buggy behavior. I think this comes down to the config vs single command override distinction - if this is config in an RC file, respecting `FLOX_SHELL` for `activate_in_place` is essentially a bug by definition

Reply:
> FLOX_SHELL means my favorite shell is SHELL, but flox doesn't support it, so when using activate_interactive, I prefer FLOX_SHELL. However, when I'm running a shell other than FLOX_SHELL and run activate_in_place, that doesn't mean I want buggy behavior. 

Matthew I see what you're saying, but the problem arises when it's flox's own shell detection that is the very thing that is buggy, and the user wants to work around that problem? In that instance we would need to introduce yet another env variable to disable the automatic shell detection to give the user control over the outcome.

With that in mind I think I agree with Yannik that the more UNIX-y behaviour is to have FLOX_SHELL as the ultimate override to all other mechanisms. That way, users have the choice of both of these outcomes:

1. running activate_in_place with FLOX_SHELL defined to a shell other than the current shell: user simply undefines FLOX_SHELL to work around the issue
2. running activate_in_place when flox's shell detection is buggy: user defines FLOX_SHELL to work around the issue

I'll just finish by saying that if we do this right it should be very rare that users define FLOX_SHELL in the first place, in which case the incidence of outcome 1 is very low, and obviously we'll work to fix the code so that outcome 2 is very low as well. Or in other words, "The Right Thing :tm:" should happen by default without the user doing anything (including the setting of FLOX_SHELL).